src/application-insights/azext_applicationinsights/_validators.py

```diff
@@ -4,11 +4,10 @@
 # --------------------------------------------------------------------------------------------
 
 # pylint: disable=len-as-condition
-from azure.cli.core.commands.client_factory import get_subscription_id
 from knack.util import CLIError
 
 
 def validate_applications(namespace):
-    if namespace.resource_group:
-        if len(apps) != 1:
+    if namespace.resource_group_name:
+        if isinstance(namespace.application, list) and len(namespace.application) != 1:
```

Comment:
This is a weird convention. The pattern in CLI is that any things referred to only by name are considered from `resource_group`. For GUIDs or ARM IDs, it would not matter. I wouldn't recommend throwing an error for this case.

Reply:
Makes sense. I was trying to avoid users mixing GUID with name/rg. If I understand correctly, querying over app1, app2 and GUID in the same resource group should be possible like this: `az monitor app-insights query -a app1 app2 GUID -g group`  but the GUID would be interpreted as a component name which will naturally fail to resolve as a resource.